### PR TITLE
Kafka Repository implementation

### PR DESCRIPTION
**Pull request by nahguam**
# stream-registry PR

# PR Checklist Forms

- [ ] CHANGELOG.md updated
- [ ] Reviewer assigned
- [ ] PR assigned (presumably to submitter)
- [ ] Labels added (enhancement, bug, documentation)

### CODE REVIEW

File: repository/kafka/src/main/java/com/expediagroup/streamplatform/streamregistry/repository/kafka/ConsumerBindingRepository.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+public class ConsumerBindingRepository
+    extends DefaultRepository<ConsumerBinding, ConsumerBindingKey, Entity.ConsumerBindingKey, DefaultSpecification>
+    implements com.expediagroup.streamplatform.streamregistry.repository.ConsumerBindingRepository {
+  ConsumerBindingRepository(EntityView view, EventSender sender, ConsumerBindingConverter converter) {
+    super(view, sender, converter, Entity.ConsumerBindingKey.class);
+  }
+
+  @Override
+  public List<ConsumerBinding> findAll(ConsumerBinding example) {
+    // This method is deprecated. It only supports the existing use case
```
Comment (rshotels) [inline repository/kafka/src/main/java/com/expediagroup/streamplatform/streamregistry/repository/kafka/ConsumerBindingRepository.java:42]: can we @Deprecate it then

File: repository/kafka/src/main/java/com/expediagroup/streamplatform/streamregistry/repository/kafka/DefaultRepository.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import com.expediagroup.streamplatform.streamregistry.state.EntityView;
+import com.expediagroup.streamplatform.streamregistry.state.EventSender;
+import com.expediagroup.streamplatform.streamregistry.state.model.Entity;
+import com.expediagroup.streamplatform.streamregistry.state.model.event.Event;
+import com.expediagroup.streamplatform.streamregistry.state.model.status.StatusEntry;
+
+@RequiredArgsConstructor
+abstract class DefaultRepository<
+    ME extends com.expediagroup.streamplatform.streamregistry.model.Entity<MK>,
+    MK,
```
Comment (rshotels) [inline repository/kafka/src/main/java/com/expediagroup/streamplatform/streamregistry/repository/kafka/DefaultRepository.java:36]: Prefer more descriptive names here

File: repository/kafka/src/main/java/com/expediagroup/streamplatform/streamregistry/repository/kafka/ConsumerBindingRepository.java (hunk 1 above)
Comment (nahguam) [reply repository/kafka/src/main/java/com/expediagroup/streamplatform/streamregistry/repository/kafka/ConsumerBindingRepository.java:42]: done

File: repository/kafka/src/main/java/com/expediagroup/streamplatform/streamregistry/repository/kafka/DefaultRepository.java (hunk 2 above)
Comment (nahguam) [reply repository/kafka/src/main/java/com/expediagroup/streamplatform/streamregistry/repository/kafka/DefaultRepository.java:42]: I've added javadoc on DefaultRepository to explain generics - good enough?

```
/**
 * @param <ME> Model Entity
 * @param <MK> Model Key
 * @param <SK> State Key
 * @param <SS> State Specification
 */
```

Comment (rshotels) [reply repository/kafka/src/main/java/com/expediagroup/streamplatform/streamregistry/repository/kafka/DefaultRepository.java:42]: Would prefer the generic names expanded, but happy with this.

**Pull request merged**